packages/react-sdk/src/expression.test.ts

```diff
@@ -145,3 +183,59 @@ test("encode/decode variable names", () => {
   );
   expect(decodeDataSourceVariable("myVarName")).toEqual(undefined);
 });
+
+test("generate effectful expression", () => {
+  expect(
+    generateEffectfulExpression(`var0 = var0 + var1`, new Set(["var0", "var1"]))
+  ).toMatchInlineSnapshot(`
+    "let var0 = _variables.get('var0');
+    let var1 = _variables.get('var1');
+    var0 = var0 + var1;
+    return new Map([
```

Comment:
just wondering why is this returning a map instead of array ['var0', var0]

Reply:
Because map has more tools

Reply:
kinda asking because having less stuff in that generated code would potentially make it easier to change over time? maybe not, I don't know.

Reply:
Just a universal api. Map for input an map for output.